sphinx/builders/__init__.py

```diff
@@ -125,6 +133,26 @@ def init(self) -> None:
         """
         pass
 
+    def merge_builder_post_transform(self, new_attrs: dict[str, Any]) -> None:
+        """Merge post-transform data into the master builder.
+
+        This method allows builders to merge any post-transform information
+        coming from parallel subprocesses back into the builder in
+        the main process. This can be useful for extensions that consume
+        that information in the build-finish phase.
+        The function is called once for each finished subprocess.
+        Builders that implement this function must also define the class
+        attribute :py:attr:`~sphinx.builders.Builder.post_transform_merge_attr`
+        as it defines which builder attributes shall be returned to
+        the main process for merging.
+
+        The default implementation does nothing.
+
+        :param new_attrs: The attributes from the parallel subprocess to be
```

Comment:
Should it be a dict or could you live with a `Mapping` (so that we really show that we do not act on the inputs).